### PR TITLE
Cherry-picked bug fixes for WP 6.5.3

### PR DESCRIPTION
**Pull request by creativecoder**
Includes

#60489 – Layout: Skip outputting base layout rules that reference content or wide sizes if no layout sizes exist
#60620 – Fix inserter pattern pagination focus loss
#60608 – Fix static posts page setting resolved template
#60641 – Font Library: Fix modal scrollbar
#60661 – Interactivity API: Allow multiple event handlers for the same type with `data-wp-on`.
#60668 – Layout: Always add semantic classes
#60845 – List View: Fix stuck dragging mode in UI in Firefox when dealing with deeply nested lists
#60764 – Don't output base flow and constrained layout rules on themes without theme.json

Also

#60686 - PHP unit test workflow: Try removing 7.0 and 7.1 to get CI tests passing

### CODE REVIEW

**Comment by creativecoder** [comment]
These three PRs are marked for backport but did not cherry-pick cleanly, I'm working on resolving the conflicts before pushing them to this branch:

- #60489
- #60668
- #60764

**Comment by creativecoder** [comment]
@andrewserong @tellthemachines Any chance you can help cherry pick https://github.com/WordPress/gutenberg/commit/59a4104691400a966e47513903afc41868b9f572 (#60489) into this branch?

There are several conflicts in `phpunit/class-wp-theme-json-test.php` and I'm not sure how they should be reconciled.

**Comment by tellthemachines** [comment]
Oh, weird that https://github.com/WordPress/gutenberg/pull/60764 was successfully cherry-picked, because it also changes the tests. Maybe the task isn't doing them in the correct order? To avoid conflicts, commits need to be cherry-picked in the order they were merged to trunk; I thought we'd fixed the task to do that 🤔 

I'll look into it!

**Comment by tellthemachines** [comment]
In any case, that commit shouldn't block the package publishing, because it's a PHP only change (it's been committed to the release branch in core already) so feel free to go ahead with it if you're pressed for time! That commit should still be cherry-picked for completeness but it's not needed here for 6.5.3 package publishing purposes.

**Comment by creativecoder** [comment]
I've manually cherry-picked the following onto this branch:

[Layout: Always add semantic classes #60668](https://github.com/WordPress/gutenberg/pull/60668) (https://github.com/WordPress/gutenberg/commit/26e35c77ffcf118d9a3f5e48c247aa0d20164904)

@Mamaduka @tellthemachines @andrewserong , I could use your help testing to make sure the fix is working correctly in this branch, as I had to manually reconcile merge conflicts. Commit to this branch is here: https://github.com/WordPress/gutenberg/pull/61299/commits/fd26a360aaeefce97b1e50b7a04b59041cf18c20

---

[Don't output base flow and constrained layout rules on themes without theme.json #60764](https://github.com/WordPress/gutenberg/pull/60764) (https://github.com/WordPress/gutenberg/commit/cc8fce2a663c9d504c900a05cecf0ec89432c6be)

The fix came in cleanly, but the test changes in `phpunit/class-wp-theme-json-test.php` did not. I manually pasted in the expected output here: https://github.com/WordPress/gutenberg/pull/61299/commits/84ac9e3f00eb25e6985c40f34333c95d7c804612#diff-88c997ea8e8ba51f2fa201327e0f1faccd4e892c2fafe33249f72cfd068c72fcR1270

@tellthemachines @oandregal @andrewserong Please take a look and make sure this fix and the test expectations are working as expected. The commit to this branch is here: https://github.com/WordPress/gutenberg/pull/61299/commits/84ac9e3f00eb25e6985c40f34333c95d7c804612

**Comment by creativecoder** [comment]
Thanks for taking a look @tellthemachines 

> Oh, weird that https://github.com/WordPress/gutenberg/pull/60764 was successfully cherry-picked, because it also changes the tests.

I had to manually resolve a conflict in the test file for #60764 as well, so that may also need a look.

> Maybe the task isn't doing them in the correct order? To avoid conflicts, commits need to be cherry-picked in the order they were merged to trunk; I thought we'd fixed the task to do that 🤔

The cherry pick script seems to be working correctly, but if a commit fails the cherry pick, it just gets skipped, so that could be a problem here. But the first PR cherry picked here was #60489, and that one failed right away. I think it's a combination of the multiple changes to `phpunit/class-wp-theme-json-test.php` since the `wp/6.5` branch was created and the long strings of expected styles that don't seem to merge cleanly.

> In any case, that commit shouldn't block the package publishing, because it's a PHP only change (it's been committed to the release branch in core already) so feel free to go ahead with it if you're pressed for time! That commit should still be cherry-picked for completeness but it's not needed here for 6.5.3 package publishing purposes.

Thanks for pointing that out, I'll keep it in mind as an option. It looks like the same is true for #60764, as well.

The plan is to create the WP 6.5.3 release candidate on May 2nd at 17:00 UTC.

**Comment by creativecoder** [comment]
@DAreRodz @luisherranz @felixarntz @gziolo It appears that an e2e test in `interactivity/deferred-store.spec.ts` is failing on this branch (https://github.com/WordPress/gutenberg/actions/runs/8917727060/job/24491222765?pr=61299).

I'm not sure if it's related to the change coming in from https://github.com/WordPress/gutenberg/pull/60661, or something else. Could you take a look?

**Comment by tellthemachines** [comment]
> it's a combination of the multiple changes to phpunit/class-wp-theme-json-test.php since the wp/6.5 branch was created

Yeah that sounds right; any other changes to the theme json class in trunk would have caused diffs in those tests. In my experience the best way to fix them is cherry-picking just the non-test changes from the related PRs, then running the tests locally and updating the expected test strings to whatever the actual output is.

Incidentally, I opened #60842 to discuss improvements to that test suite, so any thoughts you might have to add there would be welcome!

**Comment by tellthemachines** [comment]
https://github.com/WordPress/gutenberg/pull/60668 is working as expected on this PR, as is https://github.com/WordPress/gutenberg/pull/60764. 

We're still missing https://github.com/WordPress/gutenberg/pull/60489 on this branch though.

File: test/performance/specs/post-editor.spec.js [hunk 1]
```diff
@@ -657,7 +657,7 @@ test.describe( 'Post Editor Performance', () => {
 
 				const startTime = performance.now();
 
-				await page.getByRole( 'button', { name: 'Test' } ).click();
+				await page.getByText( 'Test' ).click();
```
Comment (Mamaduka) [inline test/performance/specs/post-editor.spec.js:660]: This locator was causing the failure. The element is button on `wp/6.5` branch, but after recent refactoring it's `tab` on trunk. See #60257.

**Comment by Mamaduka** [comment]
I think it's okay to ignore React Native and Interactivity e2e test failures. Latter is known to be flaky on this branch - https://github.com/WordPress/gutenberg/pull/60241#issuecomment-2023628909.

**Comment by creativecoder** [comment]
> https://github.com/WordPress/gutenberg/pull/60668 is working as expected on this PR, as is https://github.com/WordPress/gutenberg/pull/60764.

@tellthemachines Awesome, thanks for testing.

>  In my experience the best way to fix them is cherry-picking just the non-test changes from the related PRs, then running the tests locally and updating the expected test strings to whatever the actual output is.

I'll go ahead and manually cherry-pick https://github.com/WordPress/gutenberg/pull/60489, using the expected test results from the failed tests to update and get the tests passing. That feels backwards, but given this branch is effectively only for JS package releases, and the failing tests are all for PHP changes that are [separately tested by core unit tests](https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/tests/theme/wpThemeJson.php), it seems acceptable.

**Comment by creativecoder** [comment]
I've finished cherry picking "https://github.com/WordPress/gutenberg/pull/60489 – Layout: Skip outputting base layout rules that reference content or wide sizes if no layout sizes exist" to this branch.

I was able to use the git extension in VS Code to see that the changes to `phpunit/class-wp-theme-json-test.php` were limited to removing the following from the expected styles strings:

- `body .is-layout-constrained > .alignwide{max-width: var(--wp--style--global--wide-size);}`
- `max-width: var(--wp--style--global--content-size)`

This seems to line up with the intent of the PR, so I think the merge conflicts are resolved correctly.

**Comment by creativecoder** [comment]
This is now ready to merge (but still needs an approval). I'm overlooking the React Native test failures based on [this comment](https://github.com/WordPress/gutenberg/pull/61299#issuecomment-2089640260), all other CI is passing.

**Comment by aaronjorbin** [comment]
@WordPress/gutenberg-core - Is there someone that can please review this so that the bug fixes can be included in 6.5.3?

**Review by aaronjorbin — APPROVED** [review]

**Pull request merged**